<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>621</width>
    <height>730</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>GRAM</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <property name="spacing">
     <number>1</number>
    </property>
    <property name="leftMargin">
     <number>1</number>
    </property>
    <property name="topMargin">
     <number>1</number>
    </property>
    <property name="rightMargin">
     <number>1</number>
    </property>
    <property name="bottomMargin">
     <number>1</number>
    </property>
    <item>
     <widget class="TriGLWidget" name="triWidget" native="true">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>621</width>
     <height>23</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFlash">
    <property name="title">
     <string>文件(&amp;f)</string>
    </property>
    <addaction name="actionOpen"/>
    <addaction name="actionSaveAs"/>
    <addaction name="actionClose"/>
    <addaction name="separator"/>
    <addaction name="actionExit"/>
   </widget>
   <widget class="QMenu" name="menuPort">
    <property name="tearOffEnabled">
     <bool>false</bool>
    </property>
    <property name="title">
     <string>机器</string>
    </property>
    <addaction name="actionConn"/>
    <addaction name="actionDisconnect"/>
    <addaction name="separator"/>
    <addaction name="actionFirmware"/>
   </widget>
   <widget class="QMenu" name="menuHelp">
    <property name="title">
     <string>帮助</string>
    </property>
    <addaction name="actionAbout"/>
    <addaction name="actionConsole"/>
    <addaction name="actionClear_Console_2"/>
   </widget>
   <widget class="QMenu" name="menuModel">
    <property name="title">
     <string>模型</string>
    </property>
    <addaction name="actionType"/>
    <addaction name="separator"/>
    <addaction name="actionSlice"/>
    <addaction name="separator"/>
    <addaction name="actionPutOnPlatform"/>
    <addaction name="actionCenterOnPlatform"/>
    <addaction name="separator"/>
    <addaction name="actionTranslateManual"/>
    <addaction name="actionScaleManual"/>
    <addaction name="actionRotateManual"/>
    <addaction name="action_restore"/>
   </widget>
   <addaction name="menuFlash"/>
   <addaction name="menuModel"/>
   <addaction name="menuPort"/>
   <addaction name="menuHelp"/>
  </widget>
  <widget class="QStatusBar" name="status"/>
  <widget class="QDockWidget" name="dockWidget">
   <property name="sizePolicy">
    <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <property name="minimumSize">
    <size>
     <width>100</width>
     <height>126</height>
    </size>
   </property>
   <property name="windowTitle">
    <string>console</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>8</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents">
    <property name="sizePolicy">
     <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
      <horstretch>0</horstretch>
      <verstretch>0</verstretch>
     </sizepolicy>
    </property>
    <property name="minimumSize">
     <size>
      <width>0</width>
      <height>100</height>
     </size>
    </property>
    <layout class="QHBoxLayout" name="horizontalLayout">
     <property name="spacing">
      <number>5</number>
     </property>
     <property name="leftMargin">
      <number>1</number>
     </property>
     <property name="topMargin">
      <number>1</number>
     </property>
     <property name="rightMargin">
      <number>1</number>
     </property>
     <property name="bottomMargin">
      <number>1</number>
     </property>
     <item>
      <widget class="ConsoleWidget" name="log" native="true">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
      </widget>
     </item>
     <item>
      <widget class="ConsoleWidget" name="console" native="true">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QToolBar" name="toolBar">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <property name="movable">
    <bool>false</bool>
   </property>
   <property name="iconSize">
    <size>
     <width>30</width>
     <height>30</height>
    </size>
   </property>
   <property name="floatable">
    <bool>false</bool>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionOpen"/>
   <addaction name="actionSaveAs"/>
   <addaction name="actionClose"/>
   <addaction name="actionAbout"/>
   <addaction name="separator"/>
   <addaction name="actionConn"/>
   <addaction name="actionDisconnect"/>
   <addaction name="actionFirmware"/>
   <addaction name="separator"/>
   <addaction name="actionPutOnPlatform"/>
   <addaction name="actionCenterOnPlatform"/>
   <addaction name="actionRotateManual"/>
   <addaction name="actionTranslateManual"/>
   <addaction name="actionScaleManual"/>
   <addaction name="action_restore"/>
   <addaction name="separator"/>
   <addaction name="actionSlice"/>
  </widget>
  <action name="actionOpen">
   <property name="icon">
    <iconset resource="qrc.qrc">
     <normaloff>:/images/open.png</normaloff>:/images/open.png</iconset>
   </property>
   <property name="text">
    <string>打开(&amp;o)</string>
   </property>
   <property name="toolTip">
    <string>打开模型(o)</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="actionClose">
   <property name="icon">
    <iconset resource="qrc.qrc">
     <normaloff>:/images/close.png</normaloff>:/images/close.png</iconset>
   </property>
   <property name="text">
    <string>关闭(&amp;c)</string>
   </property>
   <property name="toolTip">
    <string>关闭模型(c)</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+C</string>
   </property>
  </action>
  <action name="actionExit">
   <property name="icon">
    <iconset resource="qrc.qrc">
     <normaloff>:/images/exit.png</normaloff>:/images/exit.png</iconset>
   </property>
   <property name="text">
    <string>退出(&amp;q)</string>
   </property>
   <property name="toolTip">
    <string>退出(q)</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Q</string>
   </property>
  </action>
  <action name="actionPerference">
   <property name="text">
    <string>Preference</string>
   </property>
  </action>
  <action name="actionSaveAs">
   <property name="icon">
    <iconset resource="qrc.qrc">
     <normaloff>:/images/saveas.png</normaloff>:/images/saveas.png</iconset>
   </property>
   <property name="text">
    <string>另存为(&amp;s)</string>
   </property>
   <property name="toolTip">
    <string>另存模型(s)</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="actionConn">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="qrc.qrc">
     <normaloff>:/images/connect.png</normaloff>:/images/connect.png</iconset>
   </property>
   <property name="text">
    <string>连接</string>
   </property>
  </action>
  <action name="actionFirmware">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="qrc.qrc">
     <normaloff>:/images/update.png</normaloff>:/images/update.png</iconset>
   </property>
   <property name="text">
    <string>固件更新</string>
   </property>
  </action>
  <action name="actionReboot">
   <property name="text">
    <string>Reboot</string>
   </property>
  </action>
  <action name="actionAbout">
   <property name="icon">
    <iconset resource="qrc.qrc">
     <normaloff>:/images/help.png</normaloff>:/images/help.png</iconset>
   </property>
   <property name="text">
    <string>关于 GRAM</string>
   </property>
   <property name="toolTip">
    <string>关于GRAM 3D 打印机</string>
   </property>
  </action>
  <action name="actionClear_Console">
   <property name="text">
    <string>Clear Console</string>
   </property>
  </action>
  <action name="actionQuery_status">
   <property name="text">
    <string>query status</string>
   </property>
  </action>
  <action name="actionUnlock">
   <property name="text">
    <string>unlock</string>
   </property>
  </action>
  <action name="actionLock">
   <property name="text">
    <string>lock</string>
   </property>
  </action>
  <action name="actionSet_sector_5">
   <property name="text">
    <string>set sector 5</string>
   </property>
  </action>
  <action name="actionSector_erase">
   <property name="text">
    <string>sector erase</string>
   </property>
  </action>
  <action name="actionSector_write">
   <property name="text">
    <string>sector write</string>
   </property>
  </action>
  <action name="actionSave">
   <property name="text">
    <string>Save</string>
   </property>
  </action>
  <action name="actionTranslate">
   <property name="text">
    <string>Translate</string>
   </property>
  </action>
  <action name="actionView">
   <property name="checked">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>View</string>
   </property>
  </action>
  <action name="actionPut_on_platform">
   <property name="text">
    <string>Put on platform</string>
   </property>
  </action>
  <action name="actionCenter_on_platform">
   <property name="text">
    <string>Center on platform</string>
   </property>
  </action>
  <action name="actionScale">
   <property name="text">
    <string>Scale</string>
   </property>
  </action>
  <action name="actionScaleManual">
   <property name="icon">
    <iconset resource="qrc.qrc">
     <normaloff>:/images/scale.png</normaloff>:/images/scale.png</iconset>
   </property>
   <property name="text">
    <string>缩放(&amp;a)</string>
   </property>
   <property name="toolTip">
    <string>缩放(a)</string>
   </property>
  </action>
  <action name="actionRotateManual">
   <property name="icon">
    <iconset resource="qrc.qrc">
     <normaloff>:/images/turn.png</normaloff>:/images/turn.png</iconset>
   </property>
   <property name="text">
    <string>旋转(&amp;r)</string>
   </property>
  </action>
  <action name="actionTranslateManual">
   <property name="icon">
    <iconset resource="qrc.qrc">
     <normaloff>:/images/move.png</normaloff>:/images/move.png</iconset>
   </property>
   <property name="text">
    <string>平移(&amp;t)</string>
   </property>
  </action>
  <action name="actionPutOnPlatform">
   <property name="icon">
    <iconset resource="qrc.qrc">
     <normaloff>:/images/bottom.png</normaloff>:/images/bottom.png</iconset>
   </property>
   <property name="text">
    <string>置底</string>
   </property>
  </action>
  <action name="actionCenterOnPlatform">
   <property name="icon">
    <iconset resource="qrc.qrc">
     <normaloff>:/images/center.png</normaloff>:/images/center.png</iconset>
   </property>
   <property name="text">
    <string>置中</string>
   </property>
  </action>
  <action name="actionConfig">
   <property name="text">
    <string>Config</string>
   </property>
  </action>
  <action name="actionSlice">
   <property name="icon">
    <iconset resource="qrc.qrc">
     <normaloff>:/images/Slicer.png</normaloff>:/images/Slicer.png</iconset>
   </property>
   <property name="text">
    <string>切片操作(&amp;m)</string>
   </property>
   <property name="toolTip">
    <string>打开切片操作对话框</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+M</string>
   </property>
  </action>
  <action name="actionPrint">
   <property name="text">
    <string>Print</string>
   </property>
  </action>
  <action name="actionDisconnect">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="qrc.qrc">
     <normaloff>:/images/disconnect.png</normaloff>:/images/disconnect.png</iconset>
   </property>
   <property name="text">
    <string>断开</string>
   </property>
  </action>
  <action name="actionConsole">
   <property name="text">
    <string>控制台</string>
   </property>
  </action>
  <action name="actionClear_Console_2">
   <property name="text">
    <string>清空控制台</string>
   </property>
  </action>
  <action name="action_restore">
   <property name="icon">
    <iconset resource="qrc.qrc">
     <normaloff>:/images/restore.png</normaloff>:/images/restore.png</iconset>
   </property>
   <property name="text">
    <string>返回</string>
   </property>
   <property name="toolTip">
    <string>返回原模型</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+R</string>
   </property>
  </action>
  <action name="actionType">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>机器型号</string>
   </property>
   <property name="toolTip">
    <string>选择机器型号</string>
   </property>
   <property name="shortcut">
    <string>Shift+T</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>ConsoleWidget</class>
   <extends>QWidget</extends>
   <header>consolewidget/consolewidget.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>TriGLWidget</class>
   <extends>QWidget</extends>
   <header>triglwidget.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="qrc.qrc"/>
 </resources>
 <connections/>
</ui>
